test/integration/spec/connect.js

```diff
@@ -203,6 +204,119 @@ describe('connect', function() {
     });
   });
 
+  describe.only('media region', () => {
+    let sid = null;
+    let token = null;
+    let roomName = null;
+
+    beforeEach(async () => {
+      const identity = randomName();
+      token = getToken(identity);
+      sid = await createRoom(randomName(), defaults.topology, { MediaRegion: 'gll' });
+    });
+
+    const invalidRegions = ['foo', '34', '$abc', 'abc!'];
+    const validRegions = defaults.regions ? defaults.regions.split(',') : ['au1', 'br1', 'de1', 'ie1', 'in1', 'jp1', 'sg1', 'us1', 'us2'];
+    const mediaRegions = ['without', ...invalidRegions, ...validRegions];
+
+    mediaRegions.forEach(mediaRegion => {
+      const isInvalidRegion = invalidRegions.includes(mediaRegion);
+      const mediaRegionOptions = mediaRegion === 'without' ? {} : { mediaRegion };
+
+      let scenario = 'when called ';
+      if (isInvalidRegion) {
+        scenario += `with an invalid mediaRegion : ${mediaRegion}`;
+      } else if (mediaRegion === 'without') {
+        scenario += 'without a mediaRegion ';
+      } else {
+        scenario += `with a valid mediaRegion: ${mediaRegion}`;
+      }
+
+      context(scenario, () => {
+        it(`should return a CancelablePromise that ${isInvalidRegion ? 'rejects with a MediaServerRemoteDescFailedError if the room is not p2p' : 'resolves with a Room'}`, async () => {
+          const cancelablePromise = connect(token, Object.assign({ name: sid }, mediaRegionOptions, { tracks: [] }));
+          assert(cancelablePromise instanceof CancelablePromise);
+
+          let error = null;
+          let room = null;
+          try {
+            room = await cancelablePromise;
+          } catch (error_) {
+            error = error_;
+          } finally {
+            if (room) {
+              room.disconnect();
+              await completeRoom(sid);
+            }
+            if (isInvalidRegion) {
+              if (defaults.topology !== 'peer-to-peer') {
+                assert(error instanceof MediaServerRemoteDescFailedError, `Connected to Room ${room && room.sid} with an invalid mediaRegion "${mediaRegion}"`);
+              } else {
+                assert.equal(room.mediaRegion, null);
+              }
+            } else {
+              assert.equal(error, null);
+              if (defaults.topology === 'peer-to-peer') {
+                assert.equal(room.mediaRegion, null);
+              }
+              if (defaults.topology !== 'peer-to-peer') {
+                if (['without'].includes(mediaRegion)) {
+                  assert.equal(typeof room.mediaRegion, 'string');
+                } else {
+                  assert.equal(room.mediaRegion, mediaRegion);
+                }
+              }
+              assert(room instanceof Room);
+            }
+          }
+        });
+      });
+    });
+
+    if (defaults.topology !== 'peer-to-peer') {
+      it('should allow the first participant to set the Room\'s media region', async () => {
```

Comment:
I don't think this test is required, since you're already doing this above.

Reply:
Will remove this!

Reply:
@PikaJoyce Please resolve only after pushing the changes.